src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -96,6 +96,34 @@ internal void InitializeCurl()
 
             public void EnsureResponseMessagePublished()
             {
+                // If this is the response hasn't been published yet, do any finaly processing of the response 
+                // message before it's published.
+                if (!Task.IsCompleted)
+                {
+                    // On Windows, if the response was automatically decompressed, Content-Encoding and Content-Length
+                    // headers are removed from the response. Do the same thing here.
+                    DecompressionMethods dm = _handler.AutomaticDecompression;
+                    if (dm != DecompressionMethods.None)
+                    {
+                        HttpContentHeaders contentHeaders = _responseMessage.Content.Headers;
+                        IEnumerable<string> encodings;
+                        if (contentHeaders.TryGetValues(HttpKnownHeaderNames.ContentEncoding, out encodings))
```

Comment:
Instead of stripping the headers here, why not avoid adding them to ContentHeaders in the header callback?